pkg/proofs/proofs.go

```diff
@@ -54,6 +54,12 @@ type PrefixSearchResults struct {
 	Depth       uint8
 }
 
+/*@
+pred (p PrefixSearchResults) Inv() {
+	p.Leaf != nil ==> p.Result_type == NonInclusionLeaf
```

Comment:
I guess it depends on how `PrefixSearchResults` is used, but I guess the code checks what `Result_type` is and then accesses `Leaf`. Hence, I'd enforce an invariant stating `p.Result_type == NonInclusionLeaf ==> p.Leaf.Inv()` and
```
pred (p *PrefixLeaf) Inv() {
     acc(p)
}
```

Reply:
Thanks!  Fixed in 37ba8863729f5b6c5c55eea08a1b2a4b469e8bc6 in #10.